### DISCUSSION TITLE
Builds crashing after upgrade to XCode 16.1

### DISCUSSION THREAD

**Discussion by ggirotto** [codemagic-ci-cd/codemagic-docs#2809] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [x] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

After we upgraded our workflows to use XCode 16.1 the builds stopped working.

### Expected results

We expected that the builds would succeed as expected.

### Actual results

We are facing some strange certificate issues when trying to build.

```
> ******** add-certificates --certificate /Users/builder/Library/MobileDevice/Certificates/tmpislp4eeo.p12
Add certificates to ******** /Users/builder/Library/codemagic-cli-tools/********s/31-10-24_8m7t6cgf.********-db
Searching for files matching /Users/builder/Library/MobileDevice/Certificates/tmpislp4eeo.p12
Add certificate /Users/builder/Library/MobileDevice/Certificates/tmpislp4eeo.p12 to ******** /Users/builder/Library/codemagic-cli-tools/********s/31-10-24_8m7t6cgf.********-db
security: SecKeychainItemImport: The user name or passphrase you entered is not correct.
Unable to add certificate /Users/builder/Library/MobileDevice/Certificates/tmpislp4eeo.p12 to ******** /Users/builder/Library/codemagic-cli-tools/********s/31-10-24_8m7t6cgf.********-db


Build failed :|
Failed to add certificate to ********
```

We tried to regenerate the certificates and provision on Apple portal but no success. The build always fails with this exception.

### Build id (optional)

6723f677d1cb20d750accb0e

**Comment by helinanever** [comment]
Hi @ggirotto , thanks for reporting this. We have looked into it and it seems to be a bug in the CLI tools. We're working on getting this fixed and will keep you posted.

**Reply by priitlatt** [reply, marked as answer]
@ggirotto, we've [released](https://github.com/codemagic-ci-cd/cli-tools/releases/tag/v0.54.2) a fix for this and your builds running on Xcode 16.1 VMs should work again. Sorry for the trouble.

**Reply by ggirotto** [reply]
Confirming that the fix worked. Thanks.
